explorer/ast/bindings.cpp

```diff
@@ -36,6 +36,19 @@ void Bindings::Add(Nonnull<const GenericBinding*> binding,
   }
 }
 
+void Bindings::Print(llvm::raw_ostream& out) const {
+  llvm::ListSeparator sep;
+  out << " >  bindings args: [";
+  for (const auto& [binding, value] : args_) {
+    out << sep << "`" << *binding << "`: `" << *value << "`";
+  }
+  out << "]\n >  bindings witnesses: [";
+  for (const auto& [binding, value] : witnesses_) {
+    out << sep << "`" << *binding << "`: `" << *value << "`";
+  }
+  out << "]";
+};
+
```

Comment:
See #3051: the output here will be non-deterministic, because `args_` and `witnesses_` are maps that use pointers as keys. That can result in trace tests being flaky.

We can address this by sorting the maps based on the `index()` of the bindings (for example, copy each map into a `vector`, sort by index, then print).

Reply:
I did it for `args_` using index but I am having issues with doing the same for `witness_` as there isn't index inside `ImplBinding`.
How should we handle that case?

also I was getting errors due to `index()` method as it was returning value of an optional int without checking, for now I've returned 0 in case the `index_` is `std::nullopt`, let me know if there is better way to handle it.